python/lsst/pipe/base/pipeline.py

```diff
@@ -177,7 +181,7 @@ def metadataDatasetName(self) -> Optional[str]:
         """Name of a dataset type for metadata of this task, `None` if
         metadata is not to be saved (`str`)
         """
-        if self.config.saveMetadata:
+        if cast(PipelineTaskConfig, self.config).saveMetadata:
```

Comment:
I think `self.config` needs to be a `PipelineTaskConfig`, can we just annotate it to be that to avoid `cast`?